token_bucket.go

```diff
@@ -137,6 +139,39 @@ func (tb *TokenBucket) TryToFulfill(amount Tokens) (fulfilled bool, tryAgainAfte
 	return true, 0
 }
 
+// Wait removes the given amount, waiting as long as necessary.
+func (tb *TokenBucket) Wait(amount Tokens) {
```

Comment:
Is `Wait` equivalent to `WaitCtx(context.Background(), amount)`? Is it worth expressing it like that, or are you concerned about the added cost of the context accesses?

Reply:
Good point, I removed the no-context version.

Reply:
Oh, I wasn't saying to remove it, I was just suggesting that you could implement `Wait` as `WaitCtx(context.Background(), amount)`. But this works too.

Reply:
I was thinking callers can just pass `context.Background()`. I realized that it is annoying that it returns an error in that case though, added it back.